internal/controller/apiextensions/composite/reconciler_test.go

```diff
@@ -863,3 +869,190 @@ func TestFilterToXRPatches(t *testing.T) {
 		})
 	}
 }
+
+func TestEnqueueForCompositionRevisionFunc(t *testing.T) {
+	type args struct {
+		of    resource.CompositeKind
+		list  func(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error
+		event runtimeevent.CreateEvent
+	}
+	type want struct {
+		added []interface{}
+	}
+	tests := []struct {
+		name string
+		args args
+		want want
+	}{
+		{
+			name: "empty",
+			args: args{
+				of: resource.CompositeKind{Group: "example.com", Version: "v1", Kind: "Dog"},
+				list: func(ctx context.Context, list client.ObjectList, opts ...client.ListOption) error {
+					return nil
```

Comment:
we should assert that `list` arg is of proper GVK. and that `opts` are `nil`.

Reply:
done